tests/MongoDB.Driver.Examples/InsertDataWithEncryptedFieldsExample.cs

```diff
@@ -0,0 +1,243 @@
+            var awsKmsOptions = new Dictionary<string, object>
+            {
+                { "accessKeyId", awsAccessKey },
+                { "secretAccessKey", awsSecretAccessKey }
+            };
+            kmsProviders.Add("aws", awsKmsOptions);
+            */
+
+            /* For Azure KMS, uncomment this block.
+            var azureTenantId = "<Azure account organization>";
+            var azureClientId = "<Azure client ID>";
+            var azureClientSecret = "<Azure client secret>";
+            var azureKmsOptions = new Dictionary<string, object>
+            {
+                { "tenantId", azureTenantId },
+                { "clientId", azureClientId },
+                { "clientSecret", azureClientSecret }
+            };
+            kmsProviders.Add("azure", azureKmsOptions);
+            */
+
+            /* For Gcp KMS, uncomment this block.
+            var gcpEmail = "<Gcp email>";
+            var gcpPrivateKey = "<Gcp private key>";
+            var gcpKmsOptions = new Dictionary<string, object>
+            {
+                { "email", gcpEmail },
+                { "privateKey", gcpPrivateKey }
+            };
+            kmsProviders.Add("gcp", gcpKmsOptions);
+            */
+
+            var schemaMap = new Dictionary<string, BsonDocument>();
+            schemaMap.Add(recordNamespace.ToString(), schema);
+
+            var extraOptions = new Dictionary<string, object>()
+            {
+                /* 
+                 *   uncomment the following line if you are running mongocryptd manually
+                { "mongocryptdBypassSpawn", true }
+                */
+            };
+
+            var connectionString = "mongodb://localhost:27017";
+            var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+            var autoEncryptionSettings = new AutoEncryptionOptions(
+                keyVaultNamespace: keyVaultNamespace,
+                kmsProviders: kmsProviders,
+                schemaMap: schemaMap,
+                extraOptions: extraOptions);
+            mongoClientSettings.AutoEncryptionOptions = autoEncryptionSettings;
+            return new MongoClient(mongoClientSettings);
+        }
+    }
+
+    public static class JsonSchemaCreator
+    {
+        private static readonly string DETERMINISTIC_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic";
+        private static readonly string RANDOM_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Random";
+
+        private static BsonDocument BuildEncryptMetadata(string base64KeyId)
+        {
+            var guid = GuidConverter.FromBytes(Convert.FromBase64String(base64KeyId), GuidRepresentation.Standard);
+            var binary = new BsonBinaryData(guid, GuidRepresentation.Standard);
+            return new BsonDocument("keyId", new BsonArray(new[] { binary }));
+        }
+
+        private static BsonDocument BuildEncryptedField(string bsonType, bool isDeterministic)
```

Comment:
Do we want to use the word "Build"? We use "Create" normally (like in line 210).